### PR TITLE
feat(networking): updgrade to libp2p 0.52.0

### PR DESCRIPTION
**Pull request by joshuef**
### Summary generated by Reviewpad on 26 Jun 23 04:46 UTC
This pull request updates the libp2p dependency across several files to version 0.52. The source of the dependency has also changed to a specific branch in some cases. There are various changes to the behavior of the Swarm type in `sn_networking`, including the addition of a new protocol and modifications to the local-discovery feature. Additionally, there are changes to struct implementations and import statements in several files.

### CODE REVIEW

File: sn_networking/src/event.rs [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
 use super::{
     error::{Error, Result},
-    msg::MsgCodec,
+    // msg::MsgCodec,
```
Comment (maqi) [inline sn_networking/src/event.rs]: just remove this?

File: sn_networking/src/lib.rs [hunk 2]
```diff
@@ -27,21 +27,21 @@ use self::{
     cmd::SwarmCmd,
     error::Result,
     event::NodeBehaviour,
-    msg::{MsgCodec, MsgProtocol},
+    // msg::{MsgCodec, MsgProtocol},
```
Comment (maqi) [inline sn_networking/src/lib.rs]: just remove it?

File: sn_networking/src/lib.rs [hunk 3]
```diff
@@ -255,9 +255,8 @@ impl SwarmDriver {
                 .set_request_timeout(request_response_timeout.unwrap_or(REQUEST_TIMEOUT_DEFAULT_S))
                 .set_connection_keep_alive(CONNECTION_KEEP_ALIVE_TIMEOUT);
 
-            request_response::Behaviour::new(
-                MsgCodec(),
-                iter::once((MsgProtocol(), req_res_protocol)),
+            request_response::cbor::Behaviour::new(
+                [(StreamProtocol::new("/safe/2"), req_res_protocol)],
```
Comment (maqi) [inline sn_networking/src/lib.rs]: make the string as a const?

File: sn_protocol/src/lib.rs [hunk 4]
```diff
@@ -121,7 +121,7 @@ impl NetworkAddress {
     // /// This implements the following equivalence:
     // ///
     // /// `self xor other = distance <==> other = self xor distance`
-    // pub fn for_distance(&self, d: Distance) -> libp2p::kad::kbucket::KeyBytes {
+    // pub fn for_distance(&self, d:  ) -> libp2p::kad::kbucket::KeyBytes {
```
Comment (maqi) [inline sn_protocol/src/lib.rs]: don't get the point why remove this type declare

Comment (joshuef) [reply sn_protocol/src/lib.rs]: unintentional! was due to refactor before libp2p Distance was in, I suspect. Will re-add

**Review by maqi — APPROVED** [review]

**Pull request closed without merging**